full-stack-practice/server/server.ts

```diff
@@ -0,0 +1,187 @@
+import 'dotenv/config'; // imports all the ports and env needed
+import pg from 'pg';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+export type Product = {
+  productId: number;
+  name: string;
+  imageUrl: string;
+  price: number;
+  shortDescription: string;
+  longDescription: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.get('/api/products', async (req, res, next) => {
+  try {
+    const sql = `
+      SELECT *
+        FROM "products"
+        ORDER BY "productId";
+    `;
+    const result = await db.query<Product>(sql);
+    res.json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/products/:productId', async (req, res, next) => {
+  try {
+    const productId = Number(req.params.productId);
+    if (!Number.isInteger(productId) || productId < 1) {
+      throw new ClientError(400, 'productId must be a positive integer');
+    }
+
+    const sql = `
+      SELECT *
+        FROM "products"
+        WHERE "productId" = $1;
+    `;
+    const params = [productId];
+    const result = await db.query<Product>(sql, params);
+
+    if (result.rows.length === 0) {
+      // No product found
+      return res
```

Comment:
Don't return the 404 this way. Instead, `throw new ClientError(404, message);`, like you do on line 43.